lib/src/native/realm_core.dart

```diff
@@ -421,6 +434,29 @@ class _RealmCore {
     return config.shouldCompactCallback!(totalSize, usedSize);
   }
 
+  static bool migration_callback(
+      Pointer<Void> userdata, Pointer<shared_realm> oldRealmHandle, Pointer<shared_realm> newRealmHandle, Pointer<realm_schema> schema) {
+    try {
+      final LocalConfiguration? config = userdata.toObject();
+      if (config == null) {
+        return false;
+      }
+
+      final oldSchemaVersion = _realmLib.realm_get_schema_version(oldRealmHandle);
+      final oldConfig = Configuration.local([], path: config.path, isReadOnly: true, schemaVersion: oldSchemaVersion);
+      final oldRealm = RealmInternal.getUnowned(oldConfig, RealmHandle._unowned(oldRealmHandle), isInMigration: true);
+
+      final newRealm = RealmInternal.getUnowned(config, RealmHandle._unowned(newRealmHandle), isInMigration: true);
+
+      final migration = MigrationInternal.create(oldRealm.getMigrationRealm(), newRealm, SchemaHandle.unowned(schema));
```

Comment:
Why not create the `MigrationRealm` inside the `MigrationInternal.create`. Thus we hide the `getMigrationRealm` cause right now it's available everywhere. So the signature of `MigrationInternal.create` would take two Realms and a SchemaHandle.

Reply:
`MigrationInternal` is in a different file and doesn't have access to the private constructor of `MigrationRealm`. I could have created a `MigrationRealmInternal` and expose the private constructor but that seemed like an overkill.

Reply:
oh that is in `RealmInternal` that will work. What is tripping me over is its not static. Can we make this static and pass the realm instance. Similar to `getUnowned`